conn_http_batch.go

```
@@ -0,0 +1,192 @@
+
+	// get Table columns and types
+	for r.Next() {
+		var (
+			colName string
+			colType string
+		)
+
+		err = r.Scan(&colName, &colType)
+		if err != nil {
+			return nil, err
+		}
+
+		err = block.AddColumn(colName, column.Type(colType))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &httpBatch{
+		ctx:       ctx,
+		conn:      h,
+		structMap: &structMap{},
+		block:     block,
+		query:     query,
+	}, nil
+}
+
+type httpBatch struct {
+	query     string
+	err       error
+	ctx       context.Context
+	conn      *httpConnect
+	structMap *structMap
+	sent      bool
+	block     *proto.Block
+}
+
+func (b *httpBatch) Abort() error {
+	defer func() {
+		b.sent = true
+	}()
+	if b.sent {
+		return ErrBatchAlreadySent
+	}
+	return nil
+}
+
+func (b *httpBatch) Append(v ...interface{}) error {
+	if b.sent {
+		return ErrBatchAlreadySent
+	}
+	if err := b.block.Append(v...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *httpBatch) AppendStruct(v interface{}) error {
+	values, err := b.structMap.Map("AppendStruct", b.block.ColumnsNames(), v, false)
+	if err != nil {
+		return err
+	}
+	return b.Append(values...)
+}
+
+func (b *httpBatch) Column(idx int) driver.BatchColumn {
+	if len(b.block.Columns) <= idx {
+		return &batchColumn{
+			err: &OpError{
+				Op:  "batch.Column",
+				Err: fmt.Errorf("invalid column index %d", idx),
+			},
+		}
+	}
+	return &batchColumn{
+		batch:  b,
+		column: b.block.Columns[idx],
+		release: func(err error) {
+			b.err = err
+		},
+	}
+}
+
+func (b *httpBatch) IsSent() bool {
+	return b.sent
+}
+
+func (b *httpBatch) Send() (err error) {
+	defer func() {
+		b.sent = true
+	}()
+	if b.sent {
+		return ErrBatchAlreadySent
+	}
+	if b.err != nil {
+		return b.err
+	}
+
+	r, w, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer w.Close()
+		encoder := binary.NewEncoder(w)
+		if b.block.Rows() != 0 {
+			if err = writeData(encoder, b.block, ""); err != nil {
+				fmt.Println(err)
```

can we do better than swallow these errors and print them..maybe an err channel on that we wait on and return the err at method end?

---

Fixed